File: Lab5/5.2/c_sharp/LowerCaseString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lab5._2_cSharp
{
    class LowerCaseString : MyString
    {
        private string str;

        public LowerCaseString(string str)
        {
            this.str = str.ToLower();

            for (int i = 0; i < GetLength(); i++)
            {
                if ((!(this.str[i] >= 'a') || !(this.str[i] <= 'z')))
                {
                    this.str = this.str.Remove(i, 1);
                    i--;
                }
            }
        }

        override public int GetLength() => str.Length;

        override public string Shift()
        {
           char [] temp = str.ToCharArray();
            char x = temp[0]; 
            for(int i=0; i<GetLength()-1;i++)
            {
                temp[i] = temp[i + 1];
            }
            temp[GetLength()-1] = x;
            str = new string(temp);
            return str;
        }

    }
}

File: Lab5/5.2/c++/LowerCaseString.cpp
#include "LowerCaseString.h"
using namespace std;

LowerCaseString :: LowerCaseString(string str)
{
    this->str = str;
    for (int i = 0; i < GetLength(); i++) {
        this->str[i] = towlower(this->str[i]);
    }
    for (int i = 0; i < GetLength(); i++)
    {
        if ((!(this->str[i] >= 'a') || !(this->str[i] <= 'z')))
        {
            this->str = this->str.erase(i, 1);
            i--;
        }
    }
}

    string LowerCaseString :: Shift()
    {
        char temp;
        temp = str[0];
        for (int i = 0; i < GetLength() - 1; i++)
        {
            str[i] = str[i + 1];
        }
        str[GetLength() - 1] = temp;
        return str;
    }

    int  LowerCaseString::GetLength() 
    { return size(str); }